include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1691,19 +1691,25 @@ __pattern_reverse(_ExecutionPolicy&&, _BidirectionalIterator __first, _Bidirecti
                   _IsVector _is_vector,
                   /*is_parallel=*/::std::false_type) noexcept
 {
-    __internal::__brick_reverse(__first, __last, _is_vector);
+    if (__first != __last)
```

Comment:
We dont need such check here. We do such trivial check just in parallel patterns.
Regarding to this brick - I would propose to rewrite it a little bit, like that:

```
    for (; __first != __last; ++__first)
    {
        using ::std::iter_swap;
        iter_swap(__first, --__d_last);
    }
```

Reply:
Fixed